libs/fs.js

```diff
@@ -521,13 +521,15 @@ var fs = (function() {
       from = buffer.contentSize;
     }
 
-    var newLength = (from + data.byteLength > buffer.contentSize) ? (from + data.byteLength) : (buffer.contentSize);
+    var newLength = (from + length > buffer.contentSize) ? (from + length) : (buffer.contentSize);
 
     buffer.setSize(newLength);
 
-    buffer.array.set(data, from);
+    for (var i = 0; i < length; i++) {
```

Comment:
> As in #1457, we should use set when the amount of data to write is big.

I think this is the only remaining issue here. @brendandahl can you modify the *write* function to use *TypedArray.set* when `length > 128`? That seems to be a reasonable threshold based on @mbebenita's earlier tests.